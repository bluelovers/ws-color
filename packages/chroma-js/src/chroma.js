"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.chroma = void 0;
const Color_1 = __importDefault(require("./Color"));
// @ts-ignore
exports.chroma = (...args) => {
    return new exports.chroma.Color(...args);
};
exports.chroma.Color = Color_1.default;
exports.chroma.version = '@@version';
exports.chroma.chroma = exports.chroma;
exports.chroma.default = exports.chroma;
Object.defineProperty(exports.chroma, "__esModule", { value: true });
exports.default = exports.chroma;
//# sourceMappingURL=chroma.js.map