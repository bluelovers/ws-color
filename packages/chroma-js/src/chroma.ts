import Color from './Color';
import { IChromaConstructor } from './types';

export interface chroma extends IChromaConstructor
{
	(...args): Color,
	Color: typeof Color;
	version: string;
	chroma: chroma,
	default: chroma,
}

// @ts-ignore
export const chroma: chroma = (...args) =>
{
	return new chroma.Color(...args);
};

chroma.Color = Color;
chroma.version = '@@version'

chroma.chroma = chroma;
chroma.default = chroma;

Object.defineProperty(chroma, "__esModule", { value: true });

export default chroma;
