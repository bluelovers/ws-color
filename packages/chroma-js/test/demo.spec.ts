import chroma from '../';

test(`pink.darken.saturate`, () =>
{

	let actual = chroma('pink').darken().saturate(2);

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test(`rand`, () =>
{

	let actual = chroma().rand();

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	console.dir(actual)
	expect(actual).toBeTruthy();

});

test(`empty`, () =>
{

	let actual = chroma();

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);

	expect(actual.toString()).toMatchSnapshot();
	expect(actual).toMatchSnapshot();

});
