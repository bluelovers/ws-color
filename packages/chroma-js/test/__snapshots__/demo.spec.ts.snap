// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty 1`] = `"#000000"`;

exports[`empty 2`] = `
Color {
  "_rgb": Array [
    null,
    null,
    null,
    1,
  ],
}
`;

exports[`pink.darken.saturate 1`] = `
Color {
  "_rgb": Array [
    255,
    108.82130445767132,
    147.20575158719473,
    1,
  ],
}
`;
